operations.js

```diff
@@ -0,0 +1,223 @@
+		if(parseFloat(purchase.payment) > 10000)
+			purchase.payment = '10000.000 HIVE';
+
+		try {
+			// Make the purchase
+			hive.transfer(_options.account, _options.account, purchase.payment, purchase.uid, _options.active_key)
+				.then(resolve)
+				.catch(reject);
+		} catch (err) { reject(err); }
+	});
+}
+
+async function sendDec(to, qty, account, active_key) {
+	return await sendToken(to, 'DEC', qty, account, active_key);
+}
+
+async function sendToken(to, token, qty, account = _options.account, active_key = _options.active_key) {
+	return new Promise((resolve, reject) => {
+		if(!account)
+			return reject({ error: `Error: Property "account" missing from the "options" object.` });
+
+		if(!active_key)
+			return reject({ error: `Error: Property "active_key" missing from the "options" object.` });
+
+		let data = { to, qty, token };
+
+		try {
+			hive.custom_json(`${_options.prefix}token_transfer`, data, account, active_key, true)
+				.then(resolve)
+				.catch(reject);
+		} catch (err) { reject(err); }
+	});
+}
+
+async function sendPacks(to, qty, edition, account = _options.account, active_key = _options.active_key) {
+	return new Promise((resolve, reject) => {
+		if(!_options.account)
+			return reject({ error: `Error: Property "account" missing from the "options" object.` });
+
+		if(!_options.active_key)
+			return reject({ error: `Error: Property "active_key" missing from the "options" object.` });
+
+		let data = { to, qty, edition, token: 'DEC' };
+
+		try {
+			hive.custom_json(`${_options.prefix}gift_packs`, data, account, active_key, true)
+				.then(resolve)
+				.catch(reject);
+		} catch (err) { reject(err); }
+	});
+}
+
+async function tournamentPayment(tournament_id, amount, currency, account = _options.game_account, active_key = _options.game_account_active_key) {
+	let data = { tournament_id, payment: `${amount} ${currency}` };
+
+	return new Promise((resolve, reject) => {
+		try {
+			hive.custom_json(`${_options.prefix}tournament_payment`, data, account, active_key, true)
+				.then(resolve)
+				.catch(reject);
+		} catch (err) { reject(err); }
+	});
+}
+
+async function tournamentEntry(tournament_id, player, amount, currency, signed_pw, captcha_token, account = _options.game_account, active_key = _options.game_account_active_key) {
+	let data = {
+		tournament_id,
+		player,
+		payment: `${amount} ${currency}`
+	};
+
+	if(signed_pw)
+		data.signed_pw = signed_pw;
+
+	if(captcha_token)
+		data.captcha_token = captcha_token;
+
+	return new Promise((resolve, reject) => {
+		try {
+			hive.custom_json(`${_options.prefix}enter_tournament`, data, account, active_key, true)
+				.then(resolve)
+				.catch(reject);
+		} catch (err) { reject(err); }
+	});
+}
+
+async function lookupTransaction(id, chain, client) {
+	if(chain)
+		chain = chain.toLowerCase();
+
+	return await db.lookupSingle('website.external_transactions', { id, chain }, client);
```

Comment:
await not needed on a return. Be careful of db.lookupSingle due to hiding errors.